### ISSUE TITLE
Dialog: Narrator does not read certain sections of dialog content when opened.

### ISSUE THREAD

**Issue by zelliott** [OfficeDev/office-ui-fabric-react#7150]
### Environment Information 
 
 
- __Package version(s)__: office-ui-fabric-react@6.105.1
- __Browser and OS versions__: Chrome, Edge, IE11, perhaps others. Win10 and 2016 Server.

### Please provide a reproduction of the bug in a codepen:
 

Codepen: https://codepen.io/zelliottm/pen/MzOJyx
Debug-only link for IE11: https://s.codepen.io/zelliottm/debug/MzOJyx/mWkoNxwJVdnA

#### Actual behavior:
 
Repro steps:
1. Open Narrator.
2. Use keyboard to navigate focus to the "Open Dialog" button.
3. Press [Enter].
4. **IE and Edge:** Observe that Narrator reads the dialog title, and the button description, but does not read the dialog description. **For Chrome:** Observe that Narrator only reads the button description.
 
#### Expected behavior:
 
4. I'd expect that Narrator read the dialog title, the dialog description, and the button description on all three browsers.

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? No
 
Requested priority: Normal
 
Products/sites affected: Internal

**Comment by KevinTCoughlin** [comment]
I _think_ Narrator is reading the Dialog's "header" because we apply `aria-role="header"` and `aria-level` to that `<p>` DOM node. It reads the `<button>` elements because they are focusable.

I'm unsure of the ARIA requirement regarding descriptive content such as the `secondaryText` content. I need to check with the ARIA spec before categorizing this issue. I also need to verify that one cannot read that node using Narrator's scan mode.

Thanks for the report and helpful Codepen @zelliott.

**Comment by KevinTCoughlin** [comment]
I was able to read all of the `Dialog`'s content in Edge using Narrator's Scan mode.

My steps to test were:

1. Enable Narrator.
1. Enable Scan mode. Using Windows, enabled via `CAPS + SPACE`.
1. In Edge, visit https://codepen.io/kevintcoughlin/live/GwOvZY
1. Focus "Open Dialog" button, invoke button via `ENTER`.
1. Focus shifts to `Dialog` and reads the content below. Also able to have it re-read by navigating focus in Narrator Scan mode.

```shell
Fabric Example Pen Live Preview - Microsoft Edge window, CodePen - Fabric Example Pen, Open Dialog, button,
All emails together alert dialog, Save, button,
<mark name="0 Id:14"/>alert dialog, <mark name="1 Id:14"/><prosody pitch="default"><prosody rate="default" volume="default" pitch="+20%">heading level 2</prosody></prosody> All <mark name="2 Id:14"/>emails <mark name="3 Id:14"/>together <mark name="4 Id:14"/>Your <mark name="5 Id:14"/>Inbox <mark name="6 Id:14"/>has <mark name="7 Id:14"/>changed. <mark name="8 Id:14"/>No <mark name="9 Id:14"/>longer <mark name="10 Id:14"/>does <mark name="11 Id:14"/>it <mark name="12 Id:14"/>include <mark name="13 Id:14"/>favorites, <mark name="14 Id:14"/>it <mark name="15 Id:14"/>is <mark name="16 Id:14"/>a <mark name="17 Id:14"/>singular <mark name="18 Id:14"/>destination <mark name="19 Id:14"/>for <mark name="20 Id:14"/>your <mark name="21 Id:14"/>emails. <mark name="ContinuousReadingProcessTextEnd Id:14"/>
<mark name="0 Id:15"/>Save <mark name="ContinuousReadingEnd Id:15"/>
Scan Off
Exiting Narrator
```

Here's me navigating in reverse-order using Scan to prove that the text is discoverable and read.

```shell
Cancel, button,
<mark name="0 Id:6"/>button, Save <mark name="ContinuousReadingEnd Id:6"/>
<mark name="0 Id:7"/>Your <mark name="1 Id:7"/>Inbox <mark name="2 Id:7"/>has <mark name="3 Id:7"/>changed. <mark name="4 Id:7"/>No <mark name="5 Id:7"/>longer <mark name="6 Id:7"/>does <mark name="7 Id:7"/>it <mark name="8 Id:7"/>include <mark name="9 Id:7"/>favorites, <mark name="10 Id:7"/>it <mark name="11 Id:7"/>is <mark name="12 Id:7"/>a <mark name="13 Id:7"/>singular <mark name="14 Id:7"/>destination <mark name="15 Id:7"/>for <mark name="16 Id:7"/>your <mark name="17 Id:7"/>emails. <mark name="ContinuousReadingEnd Id:7"/>
<mark name="0 Id:8"/><prosody pitch="default"><prosody rate="default" volume="default" pitch="+20%">heading level 2, </prosody></prosody> All <mark name="1 Id:8"/>emails <mark name="2 Id:8"/>together <mark name="ContinuousReadingEnd Id:8"/>
Scan Off
Exiting Narrator
```

Narrator's Scan mode is documented as providing the following utility:

>Scan mode lets you navigate apps, email, and webpages using the arrow keys. You’ll also be able to use common keyboard shortcuts to read text and jump directly to headings, links, tables, and landmarks.
>https://support.microsoft.com/en-us/help/22809/windows-10-narrator-using-scan-mode

Given the above, I lean towards closing this issue as "Won't fix" _unless_ it is causing an ARIA error / warning during validation.

Regardless of outcome, thank you for bringing this to our attention @zelliott.

**Comment by KevinTCoughlin** [comment]
@zelliott, I'm unsure what changed in the past _8 days_, but I've confirmed that Narrator reads out the `Dialog` contents in both Edge & IE 11 when _not_ in Scan mode.

## Narrator Output as Text

[Codepen used to test](https://codepen.io/kevintcoughlin/full/GwXBJx)

### Edge

```shell
CodePen - Fabric Dialog - Narrator - Microsoft Edge window, s.codepen.io, CodePen - Fabric Dialog - Narrator, 
All emails together alert dialog, Save, button,
<mark name="0 Id:5"/>alert dialog, <mark name="1 Id:5"/><prosody pitch="default"><prosody rate="default" volume="default" pitch="+20%">heading level 2</prosody></prosody> All <mark name="2 Id:5"/>emails <mark name="3 Id:5"/>together <mark name="4 Id:5"/>Your <mark name="5 Id:5"/>Inbox <mark name="6 Id:5"/>has <mark name="7 Id:5"/>changed. <mark name="8 Id:5"/>No <mark name="9 Id:5"/>longer <mark name="10 Id:5"/>does <mark name="11 Id:5"/>it <mark name="12 Id:5"/>include <mark name="13 Id:5"/>favorites, <mark name="14 Id:5"/>it <mark name="15 Id:5"/>is <mark name="16 Id:5"/>a <mark name="17 Id:5"/>singular <mark name="18 Id:5"/>destination <mark name="19 Id:5"/>for <mark name="20 Id:5"/>your <mark name="21 Id:5"/>emails. <mark name="ContinuousReadingProcessTextEnd Id:5"/>
Cancel, button,
Save, button,
```

### IE 11

```shell
CodePen - Fabric Dialog - Narrator - Internet Explorer window, CodePen - Fabric Dialog - Narrator, Open Dialog, button,
All emails together, Save, button,
<mark name="0 Id:3"/>All <mark name="1 Id:3"/>emails <mark name="2 Id:3"/>together<mark name="3 Id:3"/>Your <mark name="4 Id:3"/>Inbox <mark name="5 Id:3"/>has <mark name="6 Id:3"/>changed. <mark name="7 Id:3"/>No <mark name="8 Id:3"/>longer <mark name="9 Id:3"/>does <mark name="10 Id:3"/>it <mark name="11 Id:3"/>include <mark name="12 Id:3"/>favorites, <mark name="13 Id:3"/>it <mark name="14 Id:3"/>is <mark name="15 Id:3"/>a <mark name="16 Id:3"/>singular <mark name="17 Id:3"/>destination <mark name="18 Id:3"/>for <mark name="19 Id:3"/>your <mark name="20 Id:3"/>emails.<mark name="ContinuousReadingProcessTextEnd Id:3"/>
Cancel, button,
Save, button,
```

**Note:** To exit the Dialog, a User will tab navigate within the element and encounter both "Cancel" and "Save" buttons which are read by Narrator in both IE 11 and Edge.

## Conclusion

The OUIFR library supports Narrator across both IE 11 and Edge browsers. Unfortunately, other browsers do not have the same level of integration with Narrator.

@zelliott can you confirm Narrator's read-out for IE 11 and Edge as I have?
